bin-ranges-transfer/src/main/java/uk/gov/pay/java_lambdas/bin_ranges_transfer/App.java

```diff
@@ -68,62 +71,67 @@ public String handleRequest(Void input, final Context context) {
                 SftpClient.Handle handle = sftpClient.openDir(SFTP_DIRECTORY);
                 List<SftpClient.DirEntry> dirEntries = sftpClient.readDir(handle);
 
-                List<SftpDownload> downloadList = dirEntries.stream()
+                List<BinRangeSftpDownload> downloadList = dirEntries.stream()
                     .filter(entry -> entry.getFilename().startsWith(SFTP_FILE_PREFIX))
-                    .map(entry -> new SftpDownload(
+                    .map(entry -> new BinRangeSftpDownload(
                         String.format("%s/%s", SFTP_DIRECTORY, entry.getFilename()),
                         entry.getFilename(),
-                        extractDateString(entry.getFilename()),
                         entry.getAttributes().getSize()))
                     .toList();
-                if (!downloadList.isEmpty()) { 
-                    streamFilesToS3(sftpClient, downloadList); 
+                if (!downloadList.isEmpty()) {
+                    var candidateKey = streamFilesToS3(sftpClient, downloadList);
+                    return new Candidate(candidateKey, true);
                 } else {
                     logger.error("No BIN ranges data found on server");
                     throw new IOException();
                 }
             }
         } catch (IOException e) {
             logger.error(e.getMessage());
-            return "Error: failed to download BIN ranges from Worldpay";
+            return null;
```

Comment:
Any reason why we’re not returning an optional or just throwing an exception here?

Reply:
the stepfunction is looking for a specific value in the response to determine if the workflow should continue. we've already logged the error so it's not important to return it, hence null